### ISSUE TITLE
Use tox with to run the tests

### ISSUE THREAD

**Issue by asmeurer** [sympy/sympy-bot-old#62]
We should consider running the tests with tox, which will also alert us to things like errors with installation.

**Comment by vperic** [comment]
Yes, this sounds like a good idea. It should at least be an option. It could also be a neat way to unify the testing with multiple interpreters (to avoid the multiple reports, just with different interpreters). I'm not really familiar with how sympy-bot works, but I'll try to look at this. 

As I said in my blog post, this might be a better idea than working on setting up Jenkins. In fact, if I had any web-programming skills at all, I'd find a way to run sympy-bot on the server (and then you'd just send a command to the server, "test pull ###", and x minutes later get the results posted automatically on the pull request). Shouldn't be too hard... <shrug>

**Comment by asmeurer** [comment]
Actually, Ondrej's plan is to add a command `sympy-bot work` that would continuously test all unreviewed pull requests, and that would just run on the server.  Actually, I don't think he ever created an issue for this, so I just created issue #63 for it.

**Comment by vperic** [comment]
Well, I'd just like to lower the amount of spam produced by sympy-bot
first, as I said in: https://github.com/sympy/sympy-bot/issues/36

On Tue, Nov 1, 2011 at 1:08 AM, Aaron Meurer
reply@reply.github.com
wrote:

> Actually, Ondrej's plan is to add a command `sympy-bot work` that would continuously test all unreviewed pull requests, and that would just run on the server.  Actually, I don't think he ever created an issue for this, so I just created issue #63 for it.
> 
> ## 
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/sympy/sympy-bot/issues/62#issuecomment-2585230

## 

Vladimir Perić

**Comment by asmeurer** [comment]
Yes, we also need to find a better solution than putting a comment on the pull request for each run.  I suggested editing the OP of the pull request with a line at the bottom with the latest info (this would require a user with push access).  Or, the bot could just edit the old comments if they already exist.

**Comment by vperic** [comment]
The first suggestion sounds good (editing a line somewhere, maybe even
in the title: eg. "Tesks OK"/"Failing Tests"). I don't see requiring
push access as such a huge problem (in the worst case, if it gets
hacked or something, the changes could just get reverted). And, as
you're already aware of the "mergeability" of issues, I guess it's
currently running as a user with push access.

On Tue, Nov 1, 2011 at 1:12 AM, Aaron Meurer
reply@reply.github.com
wrote:

> Yes, we also need to find a better solution than putting a comment on the pull request for each run.  I suggested editing the OP of the pull request with a line at the bottom with the latest info (this would require a user with push access).  Or, the bot could just edit the old comments if they already exist.
> 
> ## 
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/sympy/sympy-bot/issues/62#issuecomment-2585267

## 

Vladimir Perić
